src/test/java/unit/in/ac/bits/protocolanalyzer/persistence/repository/SaveRepositoryTest.java

```diff
@@ -1,59 +1,154 @@
 package unit.in.ac.bits.protocolanalyzer.persistence.repository;
 
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
 import java.util.ArrayList;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
 
-import static org.hamcrest.CoreMatchers.equalTo;
-import static org.hamcrest.CoreMatchers.is;
-import static org.hamcrest.CoreMatchers.notNullValue;
-import static org.hamcrest.MatcherAssert.assertThat;
+import javax.naming.Context;
+import javax.naming.NamingException;
 
 import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
 import org.mockito.Mock;
 import org.mockito.MockitoAnnotations;
-import org.springframework.beans.factory.annotation.Autowired;
+import org.mockito.Spy;
+import org.mockito.runners.MockitoJUnitRunner;
 import org.springframework.data.elasticsearch.core.ElasticsearchTemplate;
 import org.springframework.data.elasticsearch.core.query.IndexQuery;
-import org.springframework.test.context.ContextConfiguration;
-import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
-import org.springframework.test.context.support.AnnotationConfigContextLoader;
 
 import com.google.common.eventbus.EventBus;
 
+import in.ac.bits.protocolanalyzer.analyzer.event.EndAnalysisEvent;
 import in.ac.bits.protocolanalyzer.persistence.repository.SaveRepository;
-import unit.config.in.ac.bits.protocolanalyzer.persistence.repository.SaveRepositoryTestConfig;
 
-@RunWith(SpringJUnit4ClassRunner.class)
-@ContextConfiguration(classes = SaveRepositoryTestConfig.class, loader = AnnotationConfigContextLoader.class)
+@RunWith(MockitoJUnitRunner.class)
 public class SaveRepositoryTest {
-	@Autowired
+	@Mock
+	public ElasticsearchTemplate template;
+	
+	@Spy
+	public ConcurrentLinkedQueue<ArrayList<IndexQuery>> buckets;
+	
+	@Mock
+	public Context ctx;
+	
+	@Mock
+	public Runtime runtime;
+	
+	@InjectMocks
 	public SaveRepository saveRepo;
+	
+	@Mock

```

Comment:
For the sake of readability, it makes sense to declare mocjs and then inject them.

Reply:
The mocks listed below "@InjectMocks" are different than the ones listed above it. The ones listed below are the mocks or dummies that are used in the Test Class, but aren't injected anywhere.

Reply:
Then the existing sequence of mock creation is alright.